summingbird-storm/src/main/scala/com/twitter/summingbird/storm/Edges.scala

```diff
@@ -113,4 +139,19 @@ private object EdgeInjections {
       (input.get(0).asInstanceOf[T1], input.get(1).asInstanceOf[T2], input.get(2).asInstanceOf[T3])
     }
   }
+
+  case class KeyValueInjection[K, V]() extends Injection[Item[(K, V)], JList[AnyRef]] {
+    override def apply(tuple: Item[(K, V)]): JList[AnyRef] = {
+      val list = new JAList[AnyRef](3)
+      list.add(tuple._1.asInstanceOf[AnyRef])
+      list.add(tuple._2._1.asInstanceOf[AnyRef])
+      list.add(tuple._2._2.asInstanceOf[AnyRef])
+      list
+    }
+
+    override def invert(valueIn: JList[AnyRef]): Try[Item[(K, V)]] = Inversion.attempt(valueIn) { input =>
+      (input.get(0).asInstanceOf[Timestamp],
```

Comment:
May be create temporary variables for readability and then
(ts, (k, v))
which is much more readable.